app/controllers/landing_page_controller.rb

```diff
@@ -93,12 +96,60 @@ def resolve_link(type, id, normalized_data)
   append_view_path "#{Rails.root}/app/views"
 
   def index
-    landing_page
+    preview_version = parse_int(params[:preview_version])
+    version = clp_version(community_id(), preview_version)
+    # TODO Ideally we would do the caching based upon just clp_version
+    # and avoid loading and parsing the (potentially) big structure
+    # JSON.
+    begin
+      structure = load_structure(community_id(), version)
+      render_landing_page(structure)
+    rescue LandingpageContentNotFound
+      render_not_found()
+    end
   end
 
   private
 
-  def landing_page
+  def parse_int(int_str_or_nil)
```

Comment:
If I get you correctly, you are aiming to parse string to integer with the exception that if the string is `nil` you want to return nil?

This can be achieved with `str&.to_i`. Now `Integer("")` throws an exception and you are forced to rescue it and return nil.